components/autosize_textarea.tsx

```diff
@@ -20,24 +21,28 @@ export class AutosizeTextarea extends React.PureComponent<Props> {
 
     private textarea?: HTMLTextAreaElement;
     private referenceRef: React.RefObject<HTMLTextAreaElement>;
+    private measuringRef: React.RefObject<HTMLDivElement>;
 
     constructor(props: Props) {
         super(props);
 
         this.height = 0;
 
         this.referenceRef = React.createRef();
+        this.measuringRef = React.createRef();
     }
 
     componentDidMount() {
-        this.recalculateSize();
+        this.recalculateHeight();
+        this.recalculateWidth();
     }
 
     componentDidUpdate() {
-        this.recalculateSize();
+        this.recalculateHeight();
+        this.recalculateWidth();
     }
```

Comment:
i know this was same as before. But should we call this calculation on any update?

Reply:
yes ... especially now since we need to support that "overflowing" behavior when the text reaches the button elements on the lower right (send, etc.)